.github/workflows/build-package-docs.yaml

```
@@ -120,3 +131,88 @@ jobs:
         name: package-docs-build
         path: build-directory/docs/docsite/ansible-package-docs-html-*.tar.gz
         retention-days: 7
+
+  deploy-package-docs:
+    if: fromJSON(github.event.inputs.deploy)
+    needs: build-package-docs
+    runs-on: ubuntu-latest
+    environment:
+      name: deploy-package-docs
+      url: https://github.com/ansible/ansible-documentation
+    env:
+      TARGET: ${{ github.event.inputs.deployment-environment }}
+      DEST_REPO: ansible-community/package-doc-builds
+      USER_EMAIL: "41898282+github-actions[bot]@users.noreply.github.com"
+      USER_NAME: "github-actions[bot]"
+    steps:
+    - name: Download the build archive
+      uses: actions/download-artifact@v4
+      with:
+        name: package-docs-build
+
+    - name: Extract the tarball
+      run: |
+        set -xeEuo pipefail
+        mkdir -pv extracted-docs
+        tar -xvzf ansible-package-docs-html-*.tar.gz -C extracted-docs
+
+    - name: Set the production branch
+      if: env.TARGET == 'production'
+      run: >-
+        echo "BRANCH=${{
+          github.event.inputs.repository-branch
+        }}" >> $GITHUB_ENV
+
+    - name: Set the test branch
+      if: env.TARGET == 'test'
+      run: echo "BRANCH=gh-pages" >> $GITHUB_ENV
+
+    - name: Checkout the deploy directory
+      uses: actions/checkout@v4
+      with:
+        repository: ${{ env.DEST_REPO }}
+        ref: ${{ env.BRANCH }}
+        path: deploy-directory
+        fetch-depth: 0
+        ssh-key: ${{ secrets.DEPLOY_DOC_BUILD }}
+        persist-credentials: true
+
+    - name: Copy the generated HTML and assets for production
+      run: >-
+        rsync -av --delete --mkpath
+        extracted-docs/
+        deploy-directory/docs
+
+    - name: Create a norobots.txt file for the test site
+      if: env.TARGET == 'test'
+      run: |
+        touch norobots.txt
+        echo "User-agent: *" > norobots.txt
+        echo "Disallow: /" >> norobots.txt
+      working-directory: deploy-directory/docs
+
+    - name: Configure the git user
+      run: |
+        git config --local user.email "${{ env.USER_EMAIL }}"
+        git config --local user.name "${{ env.USER_NAME }}"
```

Always use environment variables in shell scripts this way:
```suggestion
        git config --local user.email "${USER_EMAIL}"
        git config --local user.name "${USER_NAME}"
```
Otherwise you risk escape problems (which can turn into security problems, depending on where the content of the environment variables comes from).

---

@felixfontein Hey, thanks for the review comment. I've used that syntax for environment variables elsewhere in the workflow. I thought this would need to follow the syntax in the [`env` context](https://docs.github.com/en/actions/learn-github-actions/contexts#env-context) examples - but your suggestion looks fine when I try it on my fork.